sdk/java/pulumi/src/main/java/com/pulumi/test/Mocks.java

```diff
@@ -17,16 +18,146 @@ public interface Mocks {
      *
-     * @return A tuple of a resource identifier and resource state. State can be either a POJO
+     * @param args arguments containing resource information
+     * @return A resource identifier and resource state. State can be either a POJO
      * or a dictionary bag. The returned ID may be null for component resources.
      */
-    CompletableFuture<Tuples.Tuple2<Optional<String> /* id */, Object /* state */>> newResourceAsync(MockResourceArgs args);
+    CompletableFuture<ResourceResult> newResourceAsync(ResourceArgs args);
 
     /**
      * Invoked when the program needs to call a provider to load data (e.g., to retrieve an existing resource).
      * <p>
+     * Default implementation returns an empty map.
      *
-     * @return Invocation result, can be either a POCO or a dictionary bag.
+     * @param args arguments containing call information
+     * @return the call invocation result, can be either a POCO or a dictionary bag.
+     */
+    default CompletableFuture<Map<String, Object>> callAsync(CallArgs args) {
+        return CompletableFuture.completedFuture(Map.of());
+    }
+
+    /**
+     * ResourceArgs for use in {@link #newResourceAsync(ResourceArgs)}
      */
-    CompletableFuture<Map<String, Object>> callAsync(MockCallArgs args);
+    class ResourceArgs {
+
+        /**
+         * Resource type name
+         */
+        @Nullable
+        public final String type;
+
+        /**
+         * Resource name
+         */
+        @Nullable
+        public final String name;
+
+        /**
+         * Resource input properties
+         */
+        @Nullable
+        public final Map<String, Object> inputs;
+
+        /**
+         * A provider name
+         */
+        @Nullable
+        public final String provider;
+
+        /**
+         * Resource identifier
+         */
+        @Nullable
+        public final String id;
+
+        /**
+         * A mock resource arguments
+         *
+         * @param type     resource type name
+         * @param name     resource name
+         * @param inputs   resource input properties
+         * @param provider a provider name
+         * @param id       a resource identifier
+         */
+        public ResourceArgs(
+                @Nullable String type,
+                @Nullable String name,
+                @Nullable Map<String, Object> inputs,
+                @Nullable String provider,
+                @Nullable String id
+        ) {
+            this.type = type;
+            this.name = name;
+            this.inputs = inputs;
+            this.provider = provider;
+            this.id = id;
+        }
+    }
+
+    /**
+     * A resource identifier and resource state.
+     */
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    class ResourceResult {
+        /**
+         * A resource identifier
+         */
+        public final Optional<String> id;
+        /**
+         * A resource state
+         */
+        public final Object state;
```

Comment:
State - is that the right word to describe that this object is? 

Is it really resource outputs?